File: 2022/day08b.jl
function main()
    lines = readlines(open("input.txt"))

    width = length(lines[1])
    height = length(lines)

    trees = Array{UInt8}(undef, width, height)
    
    for (y, line) in enumerate(lines)
        for (x, chr) in enumerate(line)
            trees[y, x] = parse(UInt8, chr)
        end
    end

    max_scenic = 0

    for (x, col) in enumerate(eachcol(trees))
        for (y, row) in enumerate(eachrow(trees))
            right = row[(x + 1):end]
            down = col[(y + 1):end]
            left = row[(x - 1):-1:1]
            up = col[(y - 1):-1:1]

            scenic = 1

            for side in [right, down, left, up]
                view_dist = 0

                for tree in side
                    if trees[y, x] > tree
                        view_dist += 1
                    else
                        view_dist += 1 # Count tree that is blocking your view
                        break
                    end
                end

                scenic *= view_dist
            end

            max_scenic = max(scenic, max_scenic)
        end
    end

    println(max_scenic)
end

main()

File: 2019/Day08B.java
import java.io.*;
import java.util.Scanner;

public class Day08B {
    public static void main(String[] args) throws FileNotFoundException {
        final int WIDTH = 25;
        final int HEIGHT = 6;

        String input = new Scanner(new File("input.txt")).next();
        String image = "";

        for (int row = 0; row < HEIGHT; row++) {
            for (int col = 0; col < WIDTH; col++) {
                int idx = row * WIDTH + col;

                while (input.charAt(idx) == '2') {
                    idx += WIDTH * HEIGHT;
                }

                if (input.charAt(idx) == '0') {
                    image += ' ';
                } else {
                    image += '#';
                }
            }

            image += '\n';
        }

        System.out.print(image);
    }
}
